app/actions/remote/user.ts

```diff
@@ -332,31 +337,44 @@ export const fetchMissingProfilesByUsernames = async (serverUrl: string, usernam
     }
 };
 
-export const updateAllUsersSinceLastDisconnect = async (serverUrl: string) => {
-    const database = DatabaseManager.serverDatabases[serverUrl];
-    if (!database) {
-        return {error: `${serverUrl} database not found`};
+export const updateAllUsersSince = async (serverUrl: string, since: number, fetchOnly = false) => {
+    if (!since) {
+        return {users: []};
     }
 
-    const lastDisconnectedAt = await queryWebSocketLastDisconnected(database.database);
-
-    if (!lastDisconnectedAt) {
-        return {users: []};
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
     }
-    const currentUserId = await queryCurrentUserId(database.database);
-    const users = await queryAllUsers(database.database);
+
+    const currentUserId = await queryCurrentUserId(operator.database);
+    const users = await queryAllUsers(operator.database);
     const userIds = users.map((u) => u.id).filter((id) => id !== currentUserId);
     let userUpdates: UserProfile[] = [];
     try {
-        userUpdates = await NetworkManager.getClient(serverUrl).getProfilesByIds(userIds, {since: lastDisconnectedAt});
+        userUpdates = await NetworkManager.getClient(serverUrl).getProfilesByIds(userIds, {since});
+        if (userUpdates.length && !fetchOnly) {
+            const modelsToBatch: Model[] = [];
+            const userModels = await operator.handleUsers({users: userUpdates, prepareRecordsOnly: true});
+            modelsToBatch.push(...userModels);
+            const directChannels = await operator.database.get<ChannelModel>(CHANNEL).
```

Comment:
Nit: This variable stores both direct and gms, not only direct channels. Nevertheless, I cannot think about any other name that is not superverbose as "dmAndGMChannels".

Reply:
to me it sounds nice as is, unless you have a very strong opinion I would like to keep it like this, eventually they are all under the same category in the UI `Direct Messages`, right ?